ports/psoc6/machine_bitstream.c

```diff
@@ -0,0 +1,95 @@
+#include "py/mpconfig.h"
+#include "py/mphal.h"
+#include "FreeRTOS.h"
+#include "task.h"
+#include "py/runtime.h"
+#include "py/mphal.h"
+#include "extmod/modmachine.h"
+
+#if MICROPY_PY_MACHINE_BITSTREAM
+
+const uint32_t timing_sequence_1[] = {500, 1125, 800, 750};
+uint32_t range = 0;
+
+void machine_bitstream_high_low(mp_hal_pin_obj_t pin, uint32_t *timing_ns, const uint8_t *buf, size_t len) {
+
+    GPIO_PRT_Type *base = CYHAL_GET_PORTADDR(pin);
+    uint32_t pinNum = CYHAL_GET_PIN(pin);
+    uint32_t fcpu_mhz = mp_hal_get_cpu_freq() / 1000000;
+
+    if (memcmp(timing_ns, timing_sequence_1,  4 * sizeof(uint32_t)) == 0) {
+        range = 0;
```

Comment:
Here we should do a check of the sys frequency. Because it is not the expected the pattern won´t work, and it should exit and maybe inform about incompatible configuration.